### PR TITLE
basic implement of playwright executor

### PR DESCRIPTION
**Pull request by dev-jonghoonpark**
## Pull request description 

basic implement of playwright executor
I'm tested using pnpm. but also added npm and yarn.

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [V] tested locally
- [V] tested on cluster
- [ ] added new dependencies
- [V] updated the docs
- [V] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Comment by vsukhin** [comment]
Thank you @chooco13 ! Great job, we will checkit out carefully!

File: README.md [hunk 1]
```diff
@@ -1,77 +1,22 @@
...
 
-# Welcome to TestKube Template Executor
-
-TestKube Template Executor is a test executor skeleton for [TestKube](https://testkube.io).  
-You can use it as basic building blocks for creating a new executor.
-
-# What is an Executor?
-
-Executor is nothing more than a program wrapped into Docker container which gets JSON (testube.Execution) OpenAPI based document as an input and returns a stream of JSON output lines (testkube.ExecutorOutput), where each output line is simply wrapped in this JSON, similar to the structured logging idea. 
+# Welcome to TestKube Playwright Executor
```
Comment (vLia) [inline README.md]: Could you please update the casing of `TestKube` to `Testkube`? I see it's wrong in our templates as well, I will take care of that one :)

File: pkg/runner/playwright.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			return result, fmt.Errorf("%s zip error: %w\n\n%s", r.dependency, err, out)
+		}
+
+		directories := []string{
+			filepath.Join(projectPath, "playwright-report-zip"),
+		}
+		err := r.Scraper.Scrape(execution.Id, directories)
+		if err != nil {
+			return result.WithErrors(fmt.Errorf("scrape artifacts error: %w", err)), nil
+		}
```
Comment (vLia) [inline pkg/runner/playwright.go]: could this be extracted into a function called something like scrapeArtifacts(...) or saveReports(...)?

File: pkg/runner/playwright_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	"path/filepath"
+	"testing"
+
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+
+	cp "github.com/otiai10/copy"
+)
+
+func TestRun(t *testing.T) {
+	t.Skip("move this test to e2e test suite with valid environment setup")
```
Comment (vLia) [inline pkg/runner/playwright_test.go:15]: Would be nice to have some automated integration tests, is that possible to implement here?

**Review by vLia — APPROVED** [review]
Looks great already, excited for this contribution! 👍 
Left some minor comments, please let me know what you think.

File: pkg/runner/playwright_test.go (hunk 3 above)
Comment (dev-jonghoonpark) [reply pkg/runner/playwright_test.go:15]: Can you give an example of the tests of other 'executors' that I need to implement?
I will try to implement it with reference to examples.

What features should I test?

**Comment by nicufk** [comment]
@jfermi could you please check the docs part?

**Review by nicufk — COMMENTED** [review]
Good Job @chooco13, github actions should be updated as well, you can take as example other executors that we have, let us know if you need support with this

File: README.md [hunk 4]
```diff
@@ -1,77 +1,22 @@
...
 
+TestKube Playwright Executor is test executor for [TestKube](https://testkube.io).
```
Comment (jfermi) [inline README.md]:
```suggestion
TestKube Playwright Executor is the test executor for [TestKube](https://testkube.io).  
```

**Comment by jfermi** [comment]
> 

I made one editing change in the README. Are there other doc pages?

**Review by jfermi — COMMENTED** [review]
One editing suggestion.

File: pkg/runner/playwright_test.go (hunk 3 above)
Comment (dev-jonghoonpark) [reply pkg/runner/playwright_test.go:15]: @vLia 
I already added examples.
It was configured by referring to "testkube-executor-cypress".

Need more tests?

Comment (dev-jonghoonpark) [reply pkg/runner/playwright_test.go:15]: image

**Comment by dev-jonghoonpark** [comment]
If the default setting of 'playwright' is set, there is a phenomenon that the test does not automatically end in testkube when the test fails, so an explanation and setting code for the phenomenon have been added.
(The 'playwright' default setting opens a report if the test fails, so the process is not terminated.)

File: pkg/runner/playwright_test.go (hunk 3 above)
Comment (vLia) [reply pkg/runner/playwright_test.go:15]: Looks great now, it works on my machine as well 👍

**Comment by vLia** [comment]
> If the default setting of 'playwright' is set, there is a phenomenon that the test does not automatically end in testkube when the test fails, so an explanation and setting code for the phenomenon have been added. (The 'playwright' default setting opens a report if the test fails, so the process is not terminated.)

Is there a way to get around this in the code somehow? Like instead of opening the report, adding it as an artifact? It sounds like it's an issue in the original playwright docker image.

**Comment by dev-jonghoonpark** [comment]
@nicufk 

> Good Job @chooco13, github actions should be updated as well, you can take as example other executors that we have, let us know if you need support with this

I wrote it by referring to the cypress executor.
Please check.

**Comment by dev-jonghoonpark** [comment]
@vLia 
> Is there a way to get around this in the code somehow? Like instead of opening the report, adding it as an artifact? It sounds like it's an issue in the original playwright docker image.

That's right, this is caused by playwright's policy. 
So I added the solution to README.md.
You can fix this issue by editing the playwright config file.

**Comment by vLia** [comment]
Hey @chooco13, I managed to run some tests locally using the playwright executor with Testkube, it works well :) I will merge this one PR and will work on making this a part of Testkube (https://github.com/kubeshop/testkube/issues/1835)

**Pull request merged**